fe1-web/src/features/wallet/screens/WalletHome.tsx

```diff
@@ -1,49 +1,119 @@
 import { useNavigation } from '@react-navigation/native';
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { StyleSheet, View, ViewStyle } from 'react-native';
+import { useSelector } from 'react-redux';
 
-import { TextBlock, WideButtonView } from 'core/components';
+import { QRCode, TextBlock, WideButtonView } from 'core/components';
 import containerStyles from 'core/styles/stylesheets/containerStyles';
+import { LaoEventType } from 'features/events/objects';
+import { makeEventByTypeSelector } from 'features/events/reducer';
+import { selectCurrentLao } from 'features/lao/reducer';
+import { RollCall } from 'features/rollCall/objects';
 import STRINGS from 'resources/strings';
 
-import { WalletStore } from '../store';
+import { RollCallTokensDropDown } from '../components';
+import * as Wallet from '../objects';
+import { createDummyWalletState, clearDummyWalletState } from '../objects/DummyWallet';
+import { RollCallToken } from '../objects/RollCallToken';
 
 const styles = StyleSheet.create({
+  homeContainer: {
+    ...containerStyles.centeredXY,
+    padding: 30,
+  } as ViewStyle,
   smallPadding: {
     padding: '1rem',
   } as ViewStyle,
-  largePadding: {
-    padding: '2rem',
+  tokenSelectContainer: {
+    flexDirection: 'row',
+    marginTop: 30,
+    marginBottom: 20,
   } as ViewStyle,
 });
 
+const rollCallSelector = makeEventByTypeSelector<RollCall>(LaoEventType.ROLL_CALL);
+
 /**
- * Wallet home screen
+ * Wallet UI once the wallet is synced
  */
 const WalletHome = () => {
+  const [tokens, setTokens] = useState<RollCallToken[]>();
+  const [selectedTokenIndex, setSelectedTokenIndex] = useState(-1);
+  const [isDebug, setIsDebug] = useState(false);
+  const rollCalls = useSelector(rollCallSelector);
+  const lao = useSelector(selectCurrentLao);
+
   // FIXME: Navigation should use a defined type here (instead of any)
   const navigation = useNavigation<any>();
 
-  function importSeed() {
-    if (WalletStore.hasSeed()) {
-      navigation.navigate(STRINGS.navigation_synced_wallet);
+  useEffect(() => {
+    if (lao && rollCalls[lao.id.valueOf()]) {
+      Wallet.recoverWalletRollCallTokens(rollCalls, lao)
+        .then((rct) => {
+          if (rct.length > 0) {
+            setTokens(rct);
+            setSelectedTokenIndex(0);
+          }
+        })
+        .catch((e) => {
+          console.debug(e);
+        });
```

Comment:
In this kind of cases, where you have a clear "normal" flow and a short alternate or exception flow (e.g., showing tokens OR dealing with the lao or roll call not existing), I'd suggest the following structure to limit indentation and make your code clearer:

```suggestion
  useEffect(() => {
    if (!lao || !rollCalls[lao.id.valueOf()]) {
      // Clear tokens screen state
      setSelectedTokenIndex(-1);
      setTokens(undefined);
      return;
    }
    
    Wallet.recoverWalletRollCallTokens(rollCalls, lao)
      .then((rct) => {
        if (rct.length > 0) {
          setTokens(rct);
          setSelectedTokenIndex(0);
        }
      })
      .catch((e) => {
        console.debug(e);
      });   
```

By returning early and limiting indentation and "jumps" in code flow, it makes following the logic much easier

Reply:
True ! 👍